webclient/grpc/src/main/java/io/helidon/webclient/grpc/GrpcBaseClientCall.java

```diff
@@ -0,0 +1,209 @@
+        this.requestMarshaller = methodDescriptor.getRequestMarshaller();
+        this.responseMarshaller = methodDescriptor.getResponseMarshaller();
+    }
+
+    public Http2ClientConnection connection() {
+        return connection;
+    }
+
+    public MethodDescriptor.Marshaller<ReqT> requestMarshaller() {
+        return requestMarshaller;
+    }
+
+    public GrpcClientStream clientStream() {
+        return clientStream;
+    }
+
+    public Listener<ResT> responseListener() {
+        return responseListener;
+    }
+
+    @Override
+    public void start(Listener<ResT> responseListener, Metadata metadata) {
+        LOGGER.finest("start called");
+
+        this.responseListener = responseListener;
+
+        // obtain HTTP2 connection
+        ClientConnection clientConnection = clientConnection();
+        connection = Http2ClientConnection.create((Http2ClientImpl) grpcClient.http2Client(),
+                clientConnection, true);
+
+        // create HTTP2 stream from connection
+        clientStream = new GrpcClientStream(
+                connection,
+                Http2Settings.create(),                 // Http2Settings
+                clientConnection.helidonSocket(),       // SocketContext
+                new Http2StreamConfig() {
+                    @Override
+                    public boolean priorKnowledge() {
+                        return true;
+                    }
+
+                    @Override
+                    public int priority() {
+                        return 0;
+                    }
+
+                    @Override
+                    public Duration readTimeout() {
+                        return grpcClient.prototype().readTimeout().orElse(Duration.ofSeconds(READ_TIMEOUT_SECONDS));
+                    }
+                },
+                null,       // Http2ClientConfig
+                connection.streamIdSequence());
+
+        // start streaming threads
+        startStreamingThreads();
+
+        // send HEADERS frame
+        ClientUri clientUri = grpcClient.prototype().baseUri().orElseThrow();
+        WritableHeaders<?> headers = WritableHeaders.create();
+        headers.add(Http2Headers.AUTHORITY_NAME, clientUri.authority());
+        headers.add(Http2Headers.METHOD_NAME, "POST");
+        headers.add(Http2Headers.PATH_NAME, "/" + methodDescriptor.getFullMethodName());
+        headers.add(Http2Headers.SCHEME_NAME, "http");
+        headers.add(GRPC_CONTENT_TYPE);
+        headers.add(GRPC_ACCEPT_ENCODING);
+        clientStream.writeHeaders(Http2Headers.create(headers), false);
+    }
+
+    abstract void startStreamingThreads();
+
+    /**
+     * Unary blocking calls that use stubs provide their own executor which needs
+     * to be used at least once to unblock the calling thread and complete the
+     * gRPC invocation. This method submits an empty task for that purpose. There
+     * may be a better way to achieve this.
+     */
+    protected void unblockUnaryExecutor() {
```

Comment:
That is a strange requirement. Is there a chance we could work around this in a different way, or is it built into the `grpc` libraries?

Reply:
I couldn't find a workaround for it so far. Yes, part of the grpc libraries.